### ISSUE TITLE
Bug: (regression) `name` should be optional in RuleCreator.withoutDocs

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#11662]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```TypeScript
import { RuleCreator } from "@typescript-eslint/utils/eslint-utils";

RuleCreator.withoutDocs({
  meta: {
    messages: {},
    schema: [],
    type: 'problem',
  },
  defaultOptions: [],
  create() { return {}; }
});
```

### ESLint Config

```javascript
n/a
```

### tsconfig

```jsonc
n/a
```

### Expected Result

I expect there to be no type error

### Actual Result

type error because `name` is now required

```
Argument of type '{ meta: { messages: {}; schema: never[]; type: "problem"; }; defaultOptions: []; create(context: Readonly<RuleContext<never, []>>): {}; }' is not assignable to parameter of type 'Readonly<RuleWithMeta<[], never, unknown>>'.
  Property 'name' is missing in type '{ meta: { messages: {}; schema: never[]; type: "problem"; }; defaultOptions: []; create(context: Readonly<RuleContext<never, []>>): {}; }' but required in type 'Readonly<RuleWithMeta<[], never, unknown>>'.ts(2345)
RuleCreator.d.ts(12, 5): 'name' is declared here.
```

### Additional Info

`name` used to be optional but was made required in https://github.com/typescript-eslint/typescript-eslint/pull/11616 (breaking)

cc @y-hsgw 

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/utils`             | `8.45.0` |

**Comment by kirkwaiblinger** [comment]
Curiously, it seems like `name` is actually not an eslint core property. https://eslint.org/docs/latest/extend/custom-rules

It looks like the only meaning of `name` is to insert the rule name into the documentation with 
```ts
import { RuleCreator } from '@typescript-eslint/utils/eslint-utils';

const myPluginRuleCreator = RuleCreator(
  name => `https://my.website.org/rules/${name}`,
);

myPluginRuleCreator({
  name: 'my-rule',
  meta: {
    type: 'problem',
    docs: {
      description: 'My custom rule',
    },
    messages: {},
    schema: [],
  },
  defaultOptions: [],
  create() {
    return {};
  },
});

```

So it makes sense that `name` should be required there (as it previously was as well), but not for `withoutDocs`

**Comment by JoshuaKGoldberg** [comment]
Oh, blagh, I forgot this is how it's meant to be used.
